### DISCUSSION TITLE
watchdog restart if device it is not connected to AP - my rule failed or hw failure ?

### DISCUSSION THREAD

**Discussion by rapi3** [arendst/Tasmota#15508] [Q&A] [answered]
Hello,
I had a total power failure in a remote location and from 22 devices all running Tasmota 11.1.0.1 and 10.0.0.4 for unknown reason 4 custom wemos D1 mini running 11.1.0.1 did not succeed to come back online... stuck on boot or something else ?

A trip there 4 days later and manual power off / on was all it needed to solved the problem, all wemos D1 and Sonof RF have external power source 5v-1A.
6 Smart Power Socket restarted properly, running 10.0.0.4 and 11.1.0.1
3 Sonoff Relay restarted properly, all running 11.1.0.1
2 Sonoff RF bridges restarted properly, all running 11.1.0.1
7 Wemos D1 mini restarted properly and 4 wemos D1 mini did not, all running 11.1.0.1

I have a rule running on all devices confident that will solve wifi connection problem but it did not in this case:
This I used to restart device every hour if wifi connection it is lost:
```
ON Time#Minute|60 DO Ping3 192.168.1.1 ENDON
ON Ping#192.168.1.1#Success==0 DO restart 99 ENDON
```
Now I changed to this rule hoping that this will run better if time it is not initialized at boot, device will never connect to wifi:
```
ON System#Init DO RuleTimer8 600 ENDON
ON Rules#Timer=8 DO Backlog RuleTimer8 3660; Ping 192.168.1.1 ENDON
ON Ping#192.168.1.1#Success==0 DO restart 99 ENDON
```

Tasmota sw it is compiled with hard wifi settings.
All devices are configured with:
Fix IP address ( one of the 4 devices that did not come online used DHCP not fix IP )
wificonfig 4
wifi 3 ( changed from 4 )
wifipower 14 and 12 ( changed from 17 to prevent max power at boot )
savedata 3600 ( most of them had 0 )
...
so0 0
so36 1    ( this was supposed to fix any Boot problems ? )
so41 60  ( changed form 105 )
so56 1
so57 1
so65 1

Any ideas what I can do more to prevent this strange problem in the future ?

**Comment by juliogonzalez** [comment]
Super stupid question, just in case: Is the rule enabled?

**Reply by rapi3** [reply]
yes 100% on all devices

**Comment by rapi3** [comment]
Again another total power failure on electrical grid and exactly the same problem on exactly the same devices.
I think this 4 devices are stuck on boot and not initialized, maybe power source are faulty.

Interesting that I moved one of the CISCO AP to another channel and one of the wemos D1 was unable to switch back to the new channel like the rest of devices that was near that AP, until the wifi watchdog rule reset the device and it connected right after boot.

**Comment by rapi3** [comment, marked as answer]
Another night with a long power failure and this time I also lost contact with a Smart Socket that protect a RPI with MQTT server in that location, It had rule to delay Power On for 3 min at boot....

_I did some test on a wemos D1 with a display that run files from ufs to display data at boot I noticed that most of the time it enter in boot protection and rules are disabled._ **This explain why my wifi watchdog rule it is not working.**
I removed ufsrun files, and it boot without problems.

Disabling boot loop ( so36 0 ) it looks like the logical solution to long run tests for all devices now.
